core-bundle/src/Command/ResizeImagesCommand.php

```diff
@@ -85,181 +102,250 @@ protected function configure(): void
+                if ($timeLimit && microtime(true) - $startTime + $sleep > $timeLimit) {
+                    $this->io->writeln('Time limit of '.$timeLimit.' seconds reached.');
 
-            foreach ($processes as $index => $process) {
-                if ($process->isRunning()) {
-                    $isRunning = true;
+                    return;
                 }
 
-                // Append new output to remaining buffer from previous loop run
-                $buffers[$index] .= $process->getIncrementalOutput();
+                usleep((int) ($sleep * 1000000));
+
+                $this->table->setRows([[getmypid(), ++$count, $path]]);
+                $this->tableOutput->clear();
+                $this->table->render();
 
-                // Split output into rows
-                $rows = explode("\n", $buffers[$index]);
+                $resizeStart = microtime(true);
 
-                // Buffer and remove last line of output, as it might be incomplete
-                $buffers[$index] = array_pop($rows);
+                $this->resizeImage($path, true);
 
-                // Write remaining rows to the output with thread prefix
-                $io->write(array_map(
-                    static function ($row) use ($index) {
-                        return sprintf('%02d: ', $index + 1).preg_replace('/^.*\r/s', '', $row)."\n";
-                    },
-                    $rows
-                ));
+                $lastDuration = microtime(true) - $resizeStart;
             }
+        } finally {
+            $this->tableOutput->clear();
+            $this->io->writeln('Resized '.$count.' Images.');
+        }
+    }
 
-            usleep(15000);
-        } while ($isRunning);
+    private function executeConcurrent(float $timeLimit, float $concurrent): int
+    {
+        $startTime = microtime(true);
 
-        $io->write($buffers);
+        $phpFinder = new PhpExecutableFinder();
+        $phpPath = $phpFinder->find();
 
-        $io->write(array_map(
-            static function (Process $process): string {
-                return $process->getErrorOutput();
-            },
-            $processes
-        ));
+        $processes = array_fill(0, (int) $concurrent ?: 1, null);
+        $paths = array_fill(0, \count($processes), '');
+        $startTimes = array_fill(0, \count($processes), 0);
+        $counts = array_fill(0, \count($processes), 0);
 
-        return max(...array_map(
-            static function (Process $process): int {
-                return (int) $process->getExitCode();
-            },
-            $processes
-        ));
+        foreach ($this->storage->listPaths() as $path) {
+            while (true) {
+                if ($timeLimit && microtime(true) - $startTime > $timeLimit) {
+                    $this->io->writeln('Time limit of '.$timeLimit.' seconds reached.');
+                    break 2;
+                }
+
+                foreach ($processes as $index => $process) {
```

Comment:
Do I get this correctly? This implements a process pool of `n = concurrent` processes (if `concurrent >= 1`) and otherwise adds throttling of `m * elapsedMicroSeconds` (with `m >= 1` for `concurrent >= 0.5`, `0 < m < 1` else) between starting a new process. 

- Why does the wait time scale with the elapsed time? There has only be one parallel process that was waited on before continuing. Or am I missing something?
- Why do we need the additional `usleep(15000)`?

Reply:
> Why does the wait time scale with the elapsed time?
Because we want to distribute the CPU utilization. With `0.5` we should wait the same time as the last resize took to get an average of 50% CPU time.

> Why do we need the additional `usleep(15000)`?
So that the PHP process doesn’t go crazy while waiting on the sub processes.

> The `$startTimes` array can be omitted I guess. The single usage in L285 could imo be `$startTime` instead.
Yes, I agree.

Reply:
Fixed in 7342a2822a250b0abca29b9525fc0209fa4eaa10